src/internal/RequestFlowBuilder.ts

```
@@ -0,0 +1,191 @@
+
+    raw(script: string): ScriptBuilder {
+        this.script = script;
+        return this;
+    }
+
+    build(): string {
+        return this.script;
+    }
+}
+
+const wrapWithVariableInjectionScript = (script: string, fields: string[]): string => {
+    fields.forEach((v) => {
+        script = `
+(seq
+    (call %init_peer_id% ("${loadVariablesService}" "${loadVariablesFn}") ["${v}"] ${v})
+    ${script}
+)`;
+    });
+
+    return script;
+};
+
+const wrapWithInjectRelayScript = (script: string): string => {
+    return `
+(seq
+    (seq 
+        (call %init_peer_id% ("${loadVariablesService}" "${loadRelayFn}") [] ${relayVariableName})
+        (call %init_peer_id% ("op" "identity") [%init_peer_id%] init_peer_id)
+    )
+    ${script}
+)`;
+};
+
+export class RequestFlowBuilder {
+    private ttl: number = DEFAULT_TTL;
+    private variables = new Map<string, any>();
+    private handlerConfigs: Array<(handler: AquaCallHandler) => void> = [];
+    private buildScript: (sb: ScriptBuilder) => void;
+    private onTimeout: () => void;
+    private onError: (error: any) => void;
+
+    build() {
+        if (!this.buildScript) {
+            throw new Error();
+        }
+
+        const b = new ScriptBuilder();
+        this.buildScript(b);
+        let script = b.build();
+        script = wrapWithVariableInjectionScript(script, Array.from(this.variables.keys()));
+        script = wrapWithXor(script);
+        script = wrapWithInjectRelayScript(script);
+
+        const res = RequestFlow.createLocal(script, this.ttl);
+        res.handler.on(loadVariablesService, loadVariablesFn, (args, _) => {
+            return this.variables.get(args[0]) || {};
+        });
+        res.handler.onEvent(xorHandleService, xorHandleFn, (args) => {
+            try {
+                const msg = JSON.parse(args[0]);
+                res.raiseError(msg);
+            } catch (e) {
+                log.warn("Error handling script didn't work", e);
```

if JSON.parse throws an error should we raise an error too?